### PR TITLE
apt_repository - new option 'update_only_added' limiting apt-get update scope

### PR DESCRIPTION
**Pull request by pwilczynskiclearcode**
In order to speed up apt-get update command now apt_repository can be configured by update_only_added=yes option (default: no) that limits the update to the added repository exclusively.

That's the initial idea. @sashka have you got any suggestions?

### CODE REVIEW

**Comment by sashka** [comment]
Thank you for your contribution.

Frankly speaking `apt_repository` going to become too clever and complex and I'd try to avoid this. From my perspective `apt_repository` should be used only to add or remove entries within `sources.list.d`, and all manipulations with package cache should be left for `apt` module.

My typical use case is:
- apt_repository add ...
- apt_repository add ...
- apt_repository del ...
- fire notification "refresh apt sources"

Full refresh for all `source.list.d` rarely longs more than 10-15 seconds so I don't really care of this particular optimization. I believe that there may be other cases where updating only just added sources comes with significant benefit, but I can hardly imagine them.

File: packaging/os/apt_repository.py [hunk 1]
```diff
@@ -81,6 +87,12 @@
 # On Ubuntu target: add nginx stable repository from PPA and install its signing key.
 # On Debian target: adding PPA is not available, so it will fail immediately.
 apt_repository: repo='ppa:nginx/stable'
+
+# Call the apt-get update to get all recent package lists with their dependencies.
+apt_repository: repo='deb http://archive.canonical.com/ubuntu hardy partner' update_cache=yes
+
+# Call the apt-get update to get the recent package lists only for the added repository.
+apt_repository: repo='deb http://archive.canonical.com/ubuntu hardy partner' update_only_added=yes
```
Comment (mgedmin) [inline packaging/os/apt_repository.py:95]: It seems to me that this example will fail with "You must set update_cache=yes to use update_only_added option" if used without modification.

Comment (mgedmin) [inline packaging/os/apt_repository.py:95]: Augh, disregard that, I forgot that update_cache defaults to yes.

File: packaging/os/apt_repository.py [hunk 2]
```diff
@@ -81,6 +87,12 @@
 # On Ubuntu target: add nginx stable repository from PPA and install its signing key.
 # On Debian target: adding PPA is not available, so it will fail immediately.
 apt_repository: repo='ppa:nginx/stable'
+
+# Call the apt-get update to get all recent package lists with their dependencies.
+apt_repository: repo='deb http://archive.canonical.com/ubuntu hardy partner' update_cache=yes
```
Comment (mgedmin) [inline packaging/os/apt_repository.py:92]: Having `update_cache=yes` here misleads the reader into thinking that update_cache defaults to "no".

**Comment by gregdek** [comment]
@pwilczynskiclearcode After some review, it looks as though this PR needs some revisions. 

Please make the suggested revisions, and when you're done, please add a comment with the text "ready_for_review" and we will continue with the review process.  Thanks!

**Comment by gregdek** [comment]
@pwilczynskiclearcode A friendly reminder: this pull request has been marked as needing your action, and it hasn't been updated in the last month. 

If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we'll keep it open pending your changes.

If we receive no indication from you within a week, we will close this pull request.

**Pull request closed without merging**